docs/adrs/0009-nimbus-web-sdk-architecture.md

```diff
@@ -0,0 +1,101 @@
+* Expose a stateless API alongside the stateful API
+
+## Decision Outcome
+
+[justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force force | … | comes out best (see below)].
+
+### Positive Consequences
+
+* [e.g., improvement of quality attribute satisfaction, follow-up decisions required, …]
+* …
+
+### Negative Consequences
+
+* [e.g., compromising quality attribute, follow-up decisions required, …]
+* …
+
+## Pros and Cons of the Options
+
+### Separate `nimbus-core`, `nimbus-sdk`, and `cirrus-sdk` libraries
+
+For this option, we would take the core Nimbus business logic currently in the `nimbus-sdk` library and move it to a new `nimbus-core` library. 
+The `nimbus-sdk` and `cirrus-sdk` libraries would depend on the `nimbus-core` library, extending the functionality for their various use cases. 
+
+* Good, because separate libraries allow for strong separation of concerns.
+* Bad, because defining and using types becomes more difficult.
+  * Traits will be needed to extend functionality in the Nimbus Core library. While dynamic typing with traits is supported, it is one of the more complex features of Rust.
+* Bad, because it would result in a significant amount of code churn.
+
+### Cargo features, one library
+
+Cargo features allow for conditional compilation of libraries and library dependencies. 
+Native Rust attributes are applied to sections of Rust code to conditionally compile structs, methods, variables, and more.
+
+For more information, check out the Cargo Features reference[<sup>[1.ii]</sup>](#links) available in the Rust Cargo book.
+
+* Good, because the use of feature configurations is familiar to Rust developers.
+* Good, because all code relating to Nimbus business logic remains in one place.
+* Good, because UniFFI bindings can be conditionally generated for each feature with different UDLs.
+* Good, because dependencies can be excluded based on enabled features.
+  * As an example, RKV will only be needed for Nimbus, so the RKV dependency will only be included in compilation when the `nimbus` feature is enabled.
+* Good, because it will not increase the size of the existing Nimbus SDK binary.
+* Bad, because onboarding developers learning Rust would likely have a more difficult time understanding Cargo features.
+
+### Cargo features, separate library
+
+This is roughly the same concept as above, but the features will be used in a different way.
+Instead of features for both Nimbus and Cirrus, we would include a `stateless` feature in the Nimbus library to disable state-oriented features and dependencies.
+The Cirrus library will then exist as a separate library from the Nimbus library, primarily to function as a more visible separation of concerns between the two libraries.
+The new Cirrus library would mostly consist of the Cirrus UDL, as well as helper methods — e.g: parsing JSON received from Python before getting to the core function or other things along those lines.
```

Comment:
Yep!

Reply:
Something worth noting here is this same benefit is available in the `cargo features; one library` approach as well. Since cargo features can be used to include/exclude entire dependencies, we could introduce new dependencies for the `cirrus` feature that are not included when compiling with the `nimbus` feature.